src/nvim/CMakeLists.txt

```diff
@@ -953,3 +953,10 @@ add_custom_target(doc-eval DEPENDS ${GEN_EVAL_TOUCH})
 add_custom_target(doc-vim DEPENDS ${VIMDOC_FILES})
 add_custom_target(doc)
 add_dependencies(doc doc-vim doc-eval)
+
+add_custom_target(lintdoc
+  COMMAND ${CMAKE_COMMAND} -E env "VIMRUNTIME=${NVIM_RUNTIME_DIR}"
+    $<TARGET_FILE:nvim_bin> --clean -l 'scripts/lintdoc.lua'
+  WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+  DEPENDS nvim
```

Comment:
No, you can't do this. `nvim` isn't a file, you can't depend on it. Dependencies between targets must use `add_dependencies`.

Reply:
Quoting @dundargoc:

> - dependency from file to target or file to file can be done with `DEPENDS` in `add_custom_command`;
> - dependency from target to file is done via `DEPENDS` in `add_custom_target`;
> - dependency from target to target is done via `add_dependencies`.

Ref: https://cmake.org/cmake/help/latest/command/add_custom_command.html